beaver/worker/tail.py

```diff
@@ -236,44 +242,87 @@ def _ensure_file_is_good(self, current_time):
 
     def _run_pass(self):
         """Read lines from a file and performs a callback against them"""
-        while True:
-            try:
-                data = self._file.read(4096)
-            except IOError, e:
-                if e.errno == errno.ESTALE:
-                    self.active = False
-                    return False
+        events = []
+        line_count = 0
+        buffered_lines = 0
+        buffered_bytes = 0
+        run_start = time.time()
 
-            lines = self._buffer_extract(data)
+        try:
```

Comment:
This block is copied and pasted three times. Is it possible to set the while loop condition to while True like it was, bring this block in?

Reply:
done in latest revision.